client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -155,6 +155,10 @@ const Checkout = React.createClass( {
 			return false;
 		}
 
+		if ( this.state.previousCart.create_new_blog ) {
```

Comment:
Could we check to see if the transaction from `this.props.transaction` is complete, and bail here if it is? That would be a little more intuitive, as it isn't self-evident why we would not redirect if the previous cart had the `create_new_blog` flag.

Also, this bug can possibly happen even in the case of a normal checkout (though it'd be rarer because we don't do as much after checkout) and this change would prevent that.